src/test/java/io/lettuce/core/pubsub/PubSubCommandTest.java

```diff
@@ -75,19 +86,31 @@ class PubSubCommandTest extends AbstractRedisClientTest implements RedisPubSubLi
     String channel = "channel0";
 
     String shardChannel = "shard-channel";
+
+    String shardTestChannel = "shard-test-channel";
+
     private String pattern = "channel*";
 
     String message = "msg!";
 
+    String shardMessage = "shard msg!";
+
     @BeforeEach
     void openPubSubConnection() {
         try {
 
             client.setOptions(getOptions());
             pubsub = client.connectPubSub().async();
             pubsub.getStatefulConnection().addListener(this);
+
+            StatefulRedisClusterPubSubConnection<String, String> clusterPubSubConn = DefaultRedisClusterClient.get()
```

Comment:
I am not sure about mixing cluster and non-cluster pub/sub tests together.
I was hoping we can mark different tests with different tags so we can execute them against different harnesses - one that has cluster environment up, and another that does not.

Right now we cant tun a cluster pubsub test without having a non-cluster environment or vise versa.

Can we get these tests in another unit?

Reply:
i ll put in to a cluster test class.